llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -490,13 +492,14 @@ extractCallGraph(const Module &M, const EntryPointGroup &ModuleEntryPoints) {
   std::unique_ptr<Module> MClone = CloneModule(
       M, VMap, [&](const GlobalValue *GV) { return GVs.count(GV); });
 
-  // TODO: Use the new PassManager instead?
-  legacy::PassManager Passes;
+  ModuleAnalysisManager MAM;
+  MAM.registerPass([&] { return PassInstrumentationAnalysis(); });
```

Comment:
For the sake of my own education, could you please comment why this pass is needed?

Reply:
PassInstrumentationAnalysis pass is used by the new Pass Manager on each iteration over user specified passes.
So, it should be registered always.
Another option to register it is to call PassBuilder::registerModuleAnalyses(ModuleAnalysisManager &);

Reply:
I think that current variant is short enough, so we can leave it as-is. I just wanted to make sure that this is mandatory step